osu.Game.Rulesets.Taiko/Objects/TaikoHitObject.cs

```diff
@@ -14,7 +14,7 @@ public abstract class TaikoHitObject : HitObject
         /// <summary>
         /// Default size of a drawable taiko hit object.
         /// </summary>
-        public const float DEFAULT_SIZE = 0.45f;
+        public const float DEFAULT_SIZE = 0.475f;
```

Comment:
This is a harder value to confirm against stable. In lazer we look to be calculating this as a ratio of the playfield height, whereas in stable there's nothing like this.

I'll leave the numbers I found here in case it helps confirm this ratio:

- `hitcircle` sprite is 128px wide
- `taikohitcircle` sprite is 113px wide
- Big hits are `1.0` scale, normal hits are `0.65` (see change below).

Maybe fine to leave this variable as is with visual comparisons, but best to check how things look after applying the suggestion below.

Reply:
Tried looking for anything from these numbers, yet there still seems to be a hidden constant factor applied here.

I firstly took a screenshot on a 1024x768 resolution, wherein no scale looks to be applied at the playfield (by pasting a sprite of the playfield background to the screenshot and checking if it matches 1:1).

Afterwards I took both `taikobigcircle` and `taikobigcircleoverlay` and pasted them over the screenshot without any scale applied, then noticed they are slightly smaller than what's drawn (circle = 113px, overlay = 128px):

<img width=200 src="https://user-images.githubusercontent.com/22781491/161642730-a9a468b8-8db0-4467-9247-3a1efb9c24dc.gif" />

For roughly perfect match, the pasted sprites need to have a scale of ~114.2% applied (circle becomes ~129px, overlay becomes ~146.2px):

<img width=200 src="https://user-images.githubusercontent.com/22781491/161646900-12819fcf-f9de-47f5-a38a-7fbc7e342fbd.gif" />

Interestingly, the legacy hit target piece is also affected by this issue, and the scale values in master (0.73x for `approachcircle`, and 0.7x for `taikobigcircle`) seem like what stable actually has, since they need to be scaled with roughly the same ratio as above to display properly (which has been accidentally achieved in https://github.com/ppy/osu/pull/17621, since both `0.83 / 0.73` and `0.8 / 0.7` are equal to ~1.14, ~114%)

To summarize, only the taiko hit objects and the hit target look to be affected (~114.2% scaled from raw textures), the input drum and the playfield background display as expected. Unsure if that'll help in figuring this out, or raise more questions instead.

Reply:
That being said, excluding the deduced scale factor, the correct constants for lazer would have to be:
 - `DEFAULT_STRONG_SIZE = 128 (strong hit size) / 200 (default playfield height)`
 - `DEFAULT_SIZE = DEFAULT_STRONG_SIZE * 0.65 (normal hit scale)`